CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ mxnet_option(USE_SSE              "Build with x86 SSE instruction support" ON IF
 mxnet_option(USE_F16C             "Build with x86 F16C instruction support" ON) # autodetects support if ON
 mxnet_option(USE_LAPACK           "Build with lapack support" ON)
 mxnet_option(USE_MKL_IF_AVAILABLE "Use MKL if found" ON)
-mxnet_option(USE_MKLML_MKL        "Use MKLDNN variant of MKL (if MKL found)" ON IF USE_MKL_IF_AVAILABLE AND (NOT APPLE))
+mxnet_option(USE_MKLML_MKL        "Use MKLDNN variant of MKL (if MKL found)" ON IF USE_MKL_IF_AVAILABLE AND (NOT APPLE) AND (NOT MSVC) )
```

Comment:
`MKLML`  only rely on `MKLDNN` rather than `MKL` so we need to switch `USE_MKL_IF_AVAILABLE` to `USE_MKLDNN`. @TaoLv

Reply:
Not sure what's the `USE_MKLML_MKL` used for. But I think it's out of the scope of this PR. I hope this PR can be merged soon then we can have @yinghu5's changes for cmake including adjustments for these redundant options. @yajiedesign @szha

Reply:
Agree :)